modules/tinymce/src/core/main/ts/bookmark/GetBookmark.ts

```diff
@@ -86,24 +87,34 @@ const findIndex = (dom: DOMUtils, name: string, element: Element) => {
   return count;
 };
 
+const moveEndPointComment = (container: Node, offset: number, rng: Range, start: boolean): void => {
+  CaretFinder.firstPositionIn(container.parentElement).each((pos: CaretPosition) => {
+    const node = pos.getNode();
+    rng['set' + (start ? 'Start' : 'End')](node, offset);
+  });
+};
+
 const moveEndPoint = (rng: Range, start: boolean) => {
-  let container, offset, childNodes;
+  let container, offset;
   const prefix = start ? 'start' : 'end';
 
   container = rng[prefix + 'Container'];
   offset = rng[prefix + 'Offset'];
 
+  // normalize Table Cell selection
   if (NodeType.isElement(container) && container.nodeName === 'TR') {
-    childNodes = container.childNodes;
+    const childNodes = container.childNodes;
     container = childNodes[Math.min(start ? offset : offset - 1, childNodes.length - 1)];
     if (container) {
       offset = start ? 0 : container.childNodes.length;
       rng['set' + (start ? 'Start' : 'End')](container, offset);
     }
+  } else if (container.nodeName === '#comment') {
+    moveEndPointComment(container, offset, rng, start);
   }
 };
 
-const normalizeTableCellSelection = (rng: Range) => {
+const normalizeInvalidSelection = (rng: Range) => {
```

Comment:
This isn't really an invalid selection, it's just not an optimal selection. Maybe this should just be called `normalizeSelection` instead?

Reply:
that modify does no longer exists